core/plugins/org.polarsys.capella.core.data.fa.validation/plugin.xml

```diff
@@ -114,6 +114,25 @@
                                This rule generates a warning if a Function Port and all its connected Functional Exchanges do not have the exact same Exchange Items.
                             </description>
                          </constraint>
+                         <constraint
+                               class="org.polarsys.capella.core.data.fa.validation.functionalExchange.FunctionalExchange_loopsOnSameFunction"
+                               id="DCOM_23"
+                               isEnabledByDefault="true"
+                               lang="java"
+                               mode="Batch"
+                               name="DCOM_23 - Loops on same function check"
+                               severity="ERROR"
+                               statusCode="1">
+                            <message>
+                               {0} (Functional Exchange) {1} loops on same {2} {3}.
+                            </message>
+                            <target
+                                  class="FunctionalExchange">
+                            </target>
+                            <description>
+                               This rule generates an error if the Functional Exchange has the source and target ports defined on the same Function.
```

Comment:
can you adapt for operational analysis. there is no ports and function are called operational activity